### PR TITLE
Add simplified syntax for hash and array

### PR DESCRIPTION
**Pull request by tagomoris**
* to specify values with very simple comma (and colon for key-value separator)
 * to re-format values because comma-separated values doesn't have type information

It looks to satisfy various use-cases of 3rd party plugins, and this change is just additional
change to allow format which is currently denied as ConfigError.

### CODE REVIEW

**Comment by tagomoris** [comment]
This change is exclusive with #874

**Comment by sonots** [comment]
I prefer this than #874

File: test/config/test_types.rb [hunk 1]
```diff
@@ -124,10 +124,33 @@ class TestConfigTypes < ::Test::Unit::TestCase
 
     test 'hash' do
       assert_equal({"x"=>"v","k"=>1}, Config::HASH_TYPE.call('{"x":"v","k":1}', {}))
+      assert_equal({"x"=>"v","k"=>"1"}, Config::HASH_TYPE.call('x:v,k:1', {}))
```
Comment (sonots) [inline test/config/test_types.rb]: Could you add a test a hash with , surrounded by spaces?

Comment (tagomoris) [inline test/config/test_types.rb]: done

File: lib/fluent/config/types.rb [hunk 2]
```diff
@@ -80,19 +80,58 @@ def self.bool_value(str)
     SIZE_TYPE = Proc.new { |val, opts| Config.size_value(val) }
     BOOL_TYPE = Proc.new { |val, opts| Config.bool_value(val) }
     TIME_TYPE = Proc.new { |val, opts| Config.time_value(val) }
+
+    REFORMAT_VALUES = ->(type, value) {
```
Comment (repeatedly) [inline lib/fluent/config/types.rb]: Why `REFORMAT_VALUES`? `REFORMAT_VALUE`?

Comment (tagomoris) [inline lib/fluent/config/types.rb]: Fixed to `REFORMAT_VALUE`

**Comment by tagomoris** [comment]
Merged.

**Pull request merged**